aptos-move/aptos-gas-calibration/src/gas_meter_helpers.rs

```diff
@@ -0,0 +1,267 @@
+/// * `module_name` - Name of module
+/// * `function_name` - Name of function in the module
+pub fn sign_user_txn(
+    executor: &mut FakeExecutor,
+    module_name: &ModuleId,
+    function_name: &str,
+    iterations: u64,
+) -> u128 {
+    let elapsed = executor.exec_module(module_name, function_name, vec![], vec![], iterations);
+    println!("running time (microseconds): {}", elapsed);
+    elapsed
+}
+
+/// Publish module under user, sign and run user transaction.
+/// This runs for both Gas Meters (regular and abstract).
+///
+/// ### Arguments
+///
+/// * `package` - Built Move package
+/// * `executor` - Runs transactions
+/// * `func_identifiers` - All function names
+/// * `address` - Address associated to an account
+/// * `identifier` - Name of module
+pub fn record_gas_meter(
+    package: &BuiltPackage,
+    executor: &mut FakeExecutor,
+    func_identifiers: Vec<String>,
+    address: AccountAddress,
+    identifier: &String,
+    iterations: u64,
+) -> GasMeters {
+    // publish test-package under module address
+    let creator = executor.new_account_at(address);
+
+    let mut gas_meter = GasMeters {
+        regular_meter: Vec::new(),
+        abstract_meter: Vec::new(),
+        equation_names: Vec::new(),
+    };
+
+    // iterate over all the functions that satisfied the requirements above
+    for (sequence_num_counter, func_identifier) in func_identifiers.into_iter().enumerate() {
+        println!(
+            "Executing {}::{}::{}",
+            address,
+            identifier,
+            func_identifier.clone(),
+        );
+        gas_meter.equation_names.push(String::from(format!(
+            "{}::{}",
+            &identifier, &func_identifier
+        )));
+
+        // publish package similar to create_publish_package in harness.rs
+        println!("Signing txn for module... ");
+        let module_payload = generate_module_payload(package);
+        let counter = sequence_num_counter.try_into().unwrap();
+        sign_module_txn(executor, &creator, module_payload, counter);
+
+        // send a txn that invokes the entry function 0x{address}::{name}::benchmark
+        println!("Signing and running user txn for Regular Meter... ");
+        let module_name = get_module_name(address, identifier, &func_identifier);
+        let duration = sign_user_txn(executor, &module_name, &func_identifier, iterations);
+        gas_meter.regular_meter.push(duration);
+
+        println!("Signing and running user txn for Abstract Meter... ");
+        let gas_formula =
+            executor.exec_abstract_usage(&module_name, &func_identifier, vec![], vec![]);
+        gas_meter.abstract_meter.push(gas_formula);
+    }
+
+    gas_meter
+}
+
+/*
+ *
+ * GETTER FUNCTIONS
+ *
+ */
+/// get module name
+///
+/// ### Arguments
+///
+/// * `address` - Address associated to an account
+/// * `identifier` - Name of module
+/// * `func_identifier` - Name of function in module
+pub fn get_module_name(
```

Comment:
It seems a bit awkward to construct a `ModuleId` this way. Have you tried `ModuleId::new`?

Reply:
I'm not sure, but this is something that I saw that was used elsewhere in the codebase, so I followed that approach taken.

Reply:
Updated using `ModuleId::new` and `Identifier::new`